aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -244,6 +244,15 @@
    */
   public static final String S3_WRITE_TAGS_PREFIX = "s3.write.tags.";
 
+  /**
+   * Used by {@link S3FileIO} to tag objects when deleting. To set, we can pass a catalog property.
+   * <p>
+   * For more details, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
+   * <p>
+   * Example in Spark: --conf spark.sql.catalog.my_catalog.s3.delete.tags.my_key=my_val
```

Comment:
can use  `s3.delete.tags.my_key=my_val` as example to make it engine agnostic.

ref @rdblue [suggestion](https://github.com/apache/iceberg/pull/4334#discussion_r828195275) 
`